concord_cli/kill_task.py

```diff
@@ -1,49 +1,184 @@
 #!/usr/bin/env python
+import sys
 import thrift
 import json
 import logging
-from optparse import OptionParser
+import argparse
 from concord_cli.utils import *
+from terminaltables import AsciiTable
 
 logging.basicConfig()
 logger = logging.getLogger('cmd.kill_task')
 logger.setLevel(logging.INFO)
 
 def generate_options():
-    usage = "usage: %prog [options] arg"
-    parser = OptionParser(usage)
-    parser.add_option("--zookeeper-hosts", dest="zookeeper",
-                      action="store", help="i.e: 1.2.3.4:2181,2.3.4.5:2181")
-    parser.add_option("--task-id", dest="task_id", action="store")
-    parser.add_option("--zookeeper-path", action="store", dest="zk_path")
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-z"
+                        ,"--zookeeper-path"
+                        ,help="Path of concord topology on zk"
+                        ,default="/bolt"
+                        ,action="store")
+    parser.add_argument("-o"
+                        ,"--zookeeper-hosts"
+                        ,help="i.e: 1.2.3.4:2181,2.3.4.5:2181"
+                        ,default="localhost:2181"
+                        ,action="store")
+    parser.add_argument("-t"
+                        ,"--task_id"
+                        ,help="The id of task to kill"
+                        ,action="store")
+    parser.add_argument("-a"
+                        ,"--all"
+                        ,help="Kill all computations"
+                        ,action="store_true")
     return parser
 
-def validate_options(options, parser):
-    if not options.task_id: parser.error("please specify --task-id")
-    if not options.zk_path: parser.error("please specify --zookeeper-path")
-    if not options.zookeeper: parser.error("please specify --zookeeper-hosts")
-
-
-def kill(options):
+def kill(zookeeper, zk_path, *task_ids):
```

Comment:
any reason we splat task_ids? since we're not really exposing this to other people i think it's fine to have it just be a regular list